include/sys/ddt_impl.h

```diff
@@ -62,7 +66,7 @@ extern void ddt_stat_update(ddt_t *ddt, ddt_entry_t *dde, uint64_t neg);
  * outside of the DDT implementation proper, and if you do, consider moving
  * them up.
  */
-#define	DDT_NAMELEN	107
+#define	DDT_NAMELEN	110
```

Comment:
Yes, adding `ddt_op_contains` grows the struct enough to trip this:

```
module/zfs/ddt.c: In function 'ddt_object_name':
./include/sys/dmu.h:377:41: error: '%s' directive output may be truncated writing up to 103 bytes into a region of size 102 [-Werror=format-truncation=]
  377 | #define DMU_POOL_DDT                    "DDT-%s-%s-%s"
      |                                         ^~~~~~~~~~~~~~
module/zfs/ddt.c:284:44: note: in expansion of macro 'DMU_POOL_DDT'
  284 |         (void) snprintf(name, DDT_NAMELEN, DMU_POOL_DDT,
      |                                            ^~~~~~~~~~~~
./include/sys/dmu.h:377:49: note: format string is defined here
  377 | #define DMU_POOL_DDT                    "DDT-%s-%s-%s"
      |                                                 ^~
module/zfs/ddt.c:284:16: note: 'snprintf' output 7 or more bytes (assuming 110) into a destination of size 107
  284 |         (void) snprintf(name, DDT_NAMELEN, DMU_POOL_DDT,
      |                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  285 |             zio_checksum_table[ddt->ddt_checksum].ci_name,
      |             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  286 |             ddt_ops[type]->ddt_op_name, ddt_class_name[class]);
      |             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

I don't fully understand it. Even though we're only expanding `ddt_op_name`, it seems to be imagining that that it might run to the end of the entire struct, not just that field. I assume its some obscure struct layout thing.

Reply:
That id strange, but I'm glad you caught it.  Thanks for confirming.